### PR TITLE
New data

### PR DESCRIPTION
**Pull request by ggreif**
This is a minimal first draft. Please don't merge yet.

Stuff missing:
* doctests
* usage

I am happy to bikeshed a bit about naming.

### CODE REVIEW

**Comment by christiaanb** [comment]
Thanks for the initial patch.

Some comments:
* I'd like this function to be in the `CLaSH.Prelude.BlockRam` module, not in a seperate module. I know the `File` submodules are there, but I did that out of `SafeHaskell` considerations. It can be put under the (haddock) section _Read/Write conflict resolution_.
* Aside from adding this function, I'd like the haddocks of the other `blockRam` functions to reference this function.
* Currently, the function is specialised on `Signal`; I'd like it to be polymorphic in the clock domain. So every occurrence of `Signal` would have to be replaced by `Signal' clk`.

**Comment by ggreif** [comment]
Okay, I tried to respond to your comments, please re-review.

**Comment by ggreif** [comment]
@christiaanb Haddockc and a doctest are in place.

File: src/CLaSH/Prelude/BlockRam.hs [hunk 1]
```diff
@@ -528,3 +536,22 @@ blockRam# clk content wr rd en din = register' clk undefined dout
...
       return d'
+
+-- | Create read-after-write blockRAM from a read-before-write one (synchronised to specified clock)
+--
+readNew' :: Eq addr => SClock clk -> (Signal' clk addr -> Signal' clk addr -> Signal' clk Bool -> Signal' clk a -> Signal' clk a) -> Signal' clk addr -> Signal' clk addr -> Signal' clk Bool -> Signal' clk a -> Signal' clk a
+readNew' clk ram wrAddr rdAddr wrEn wrData = mux wasSame wasWritten $ ram wrAddr rdAddr wrEn wrData
+  where sameAddr = (==) <$> wrAddr <*> rdAddr
+        wasSame = register' clk False ((&&) <$> wrEn <*> sameAddr)
+        wasWritten = register' clk undefined wrData
+        mux = liftA3 (\b t f -> if b then t else f)
```
Comment (christiaanb) [inline src/CLaSH/Prelude/BlockRam.hs]: `mux` already exists in [`CLaSH.Signal`](https://github.com/clash-lang/clash-prelude/blob/master/src/CLaSH/Signal.hs#L20).

**Comment by ggreif** [comment]
@christiaanb Sorry I forgot that line from a debugging session. Fixed now.

**Comment by christiaanb** [comment]
OK, looks good to me. Thanks for the patch.

**Pull request merged**